documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,28 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 -->
-# StateSynchronizer Design
+# State Synchronizer Design
 
-A StateSynchronizer provides a means for data to be written and read by multiple processes, while consistency is guaranteed using optimistic checks. A rough version of this API is [checked in here.](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java)
+A State Synchronizer provides a means for data to be written and read by multiple processes, while consistency is guaranteed using optimistic checks. For more information, check the [API](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java)
 
-This works by having each process keep a copy of the data. All updates are written through the StateSynchronizer which appends them to a Pravega segment. They can keep up to date of changes to the data by consuming from the segment. To provide consistency a conditional append is used. This ensures that the updates can only proceed if the process performing them has the most recent data. Finally to prevent the segment from growing without bound, we use have a compact operation that re-writes the latest data, and truncates the old data so it can be dropped.
+This works by having each process keep a copy of the data. All the updates are written through the State Synchronizer which appends them to the Pravega Stream Segment. Latest updates can be incorporated to the data by consuming from the Stream Segment. To provide consistency a conditional append is used. This ensures that the updates can only proceed if the process performing them has the most recent data. To avoid the unbounded data in the Stream Segment, a compact operation is involved which re-writes the latest data and truncates the old data.
```

Comment:
I think that what is summarized as a single paragraph deserves to be explained in much more detail. This is a design/implementation document, so here is where the actual technical decisions and details should appear and be justified, so users can understand why the State Synchronizer works as it does. 

Actually, reviewing the documents related to State Synchronizer we have:
- Introductory section in Concepts: http://pravega.io/docs/latest/pravega-concepts/#state-synchronizers
- Working with State Synchronizer: http://pravega.io/docs/latest/state-synchronizer/ (by the way, I think that the URL is misleading) 

And in both documents, I see more detailed explanations than in this document, despite this is the one that should really describe in depth the internals of this mechanism. If possible, I suggest to extend this document, which will help users to approach this non-trivial abstraction.

Reply:
@tkaitchuck need ur opinion on this!

Reply:
Yes. Raul is correct. Take a look over those two linked docs, and any details they mention as part of this, we can duplicate here.

Reply:
@tkaitchuck @RaulGracia Accepted suggestion